### PR TITLE
feat(): add PWA support

### PR DESCRIPTION
**Pull request by jgw96**

### CODE REVIEW

File: angular/base/package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@angular/platform-browser": "6.0.0",
     "@angular/platform-browser-dynamic": "6.0.0",
     "@angular/router": "6.0.0",
+    "@angular/service-worker": "^6.0.0",
```
Comment (dwieeb) [inline angular/base/package.json]: Let's be precise with Angular dependencies. `^6.0.0` grabs a scary amount of version ranges

File: angular/base/ngsw-config.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (dwieeb) [inline angular/base/ngsw-config.json]: Can this file go in `src/`?

Comment (mhartington) [reply angular/base/ngsw-config.json]: No, as far as I can tell, it needs to be in the root dir

File: angular/base/package.json (hunk 1 above)
Comment (dwieeb) [reply angular/base/package.json]: keywords being "should not" 

It's not a good experience to ship starters with these version ranges and having it break if a mistake was made on Angular's end. We're on cutting edge technology, so it's safer to lock in versions and test the starters before deploying them to everyone.

Comment (dwieeb) [reply angular/base/package.json]: Plus, we use dependabot now, so upgrades to latest versions happen quite quickly, as long as CI is passing

**Comment by dwieeb** [comment]
Closing for now as was our decision in slack.

Gist:

Write the PWA docs with this workflow for now: 1) `ng add @angular/pwa` 2) manually remove `serviceWorker` entries in `angular.json` 3) build PWA with `ionic build -- --service-worker`

**Pull request closed without merging**